rpc/gnmi/gnmi-history.md

```diff
@@ -0,0 +1,87 @@
+# gNMI History Extension
+
+**Contributors:** Justin Costa-Roberts, Aaron Beitch, Mike Fink
+
+**Date:** August 4, 2020
+
+**Version:** 0.1.0
+
+# 1. Purpose
+
+In certain applications it is necessary not only to retrieve current data from a
+data tree, but also to allow querying historical data. The gNMI History
+extension enables clients both to retrieve data at a specific time in the past,
+and to request all updates applied to a data tree between two specified times.
+
+# 2. Definition
+
+A `History` message is embedded in an Extension message in the
+`SubscribeRequest` proto.
+
+## 2.1 Proto
+
+```
+message TimeRange {
+  int64 start = 1;
+  int64 end = 2;
+}
+
+message History {
+  oneof request {
+    int64 time = 1;
```

Comment:
```proto
message History {
  oneof request {
    int64 timestamp = 1;
    TimeRange range = 2;
  }
}
```

might clearer on what the timestamp represents - when reading this the first time i thought it was a start without end vs. a stream or snapshot

Reply:
Maybe even `at_timestamp` or `snapshot_timestamp` would be clearer. I agree with you that coming back this doesn't look as clear as it could do.

Reply:
I've changed the field name here (and in gnmi_ext.proto) to `snapshot_time`. I like `at_time` better as a standalone name but making one field a prepositional phrase and the other a noun kind of stresses me out. And went with `time` over `timestamp` because to me "timestamp" indicates a time at which something _happened_, whereas the client here isn't indicating anything happening at all--it just wants the tree at the specified time. Let me know if you have any objection.